apps/fz_http/priv/repo/migrations/20230116200524_add_not_null_to_mfa_methods.exs

```diff
@@ -0,0 +1,13 @@
+defmodule FzHttp.Repo.Migrations.AddNotNullToMfaMethods do
+  use Ecto.Migration
+
+  def change do
+    alter table("mfa_methods") do
+      remove(:credential_id, :string)
+      modify(:payload, :bytea, null: false)
```

Comment:
Hm, think this might break existing instances where these fields may not be set?

Reply:
@jamilbk there can't exist an MFA method without a `payload`, it's the field that keeps the TOTP `secret`. Same with `last_used_at`, if anyone has that - their installation already doesn't work.

We might want to unlock those by setting a default value for last_used_at and removing MFA methods without a payload, WDYT?

Reply:
Not worried about the MFA breaking, but moreso about the migration itself failing due to changing the NULL constraint on existing data with NULL fields.

Hmm, maybe fill `payload` with random bytes and set `last_used_at` to the epoch?

Reply:
Okay, but with random bytes we won't abe able to decode encoded field (payload is using `Encrypted.Map`) so following exception will be raised:
```
** (ArgumentError) cannot load `"FjnES7y1R4GUkkWVIiym6m/a3IcDkA0+lhCOkz9fc2Q="` as type FzHttp.Encrypted.Map for field :payload in %FzHttp.MFA.Method{__meta__: #Ecto.Schema.Metadata<:loaded, "mfa_methods">, id: nil, name: nil, type: nil, last_used_at: nil, payload: nil, code: nil, user_id: nil, user: #Ecto.Association.NotLoaded<association :user is not loaded>, inserted_at: nil, updated_at: nil}
```

I guess that's still better than auto-disabling MFA when config is invalid.

Reply:
Hmm, I guess we could `DROP` the offending records then.

Reply:
@jamilbk I initially intended to DROP them but felt like it's a bad security practice because something locked forever with a broken MFA will be unlocked after migration and maybe someone can run into trouble because of that.. Anyway, I hope there are not a lot of people that have broken configs in DB.